beets/util/functemplate.py

```diff
@@ -512,6 +513,18 @@ def _parse_ident(self):
         return ident
 
 
+class ArgumentsParser(Parser):
+    """``Parser`` that considers ``ARG_SEP`` to be a special character.
+    """
+    # Common parsing resources.
+    special_chars = (SYMBOL_DELIM, FUNC_DELIM, GROUP_OPEN, GROUP_CLOSE,
+                     ARG_SEP, ESCAPE_CHAR)
+    special_char_re = re.compile(r'[%s]|$' %
+                                 u''.join(re.escape(c) for c in special_chars))
+    escapable_chars = (SYMBOL_DELIM, FUNC_DELIM, GROUP_CLOSE, ARG_SEP)
+    terminator_chars = (GROUP_CLOSE, ARG_SEP)
```

Comment:
I like this subclassing approach, but I'm a little sad that it required us to copy n' paste the definitions from the base class. This seems like it could get us into trouble if we eventually change one set of lists and forget to update the other one.

It's a tough call, but maybe just an `in_argument` flag on `Parser` would be simpler?